kafka/producer/producer_batch.go

```diff
@@ -136,3 +150,28 @@ func isFatalError(err error) bool {
 	}
 	return true
 }
+
+func (b *Batch) handleWriteError(writeErrors kafka.WriteErrors) {
+	for i := range writeErrors {
+		if writeErrors[i] != nil {
+			b.callback.OnError(convertKafkaMessage(b.messages[i]))
+		} else {
+			b.callback.OnSuccess(convertKafkaMessage(b.messages[i]))
+		}
+	}
+}
+
+func (b *Batch) handleCallbackSuccess() {
+	for _, msg := range b.messages {
+		b.callback.OnSuccess(convertKafkaMessage(msg))
+	}
+}
+
+func convertKafkaMessage(src kafka.Message) message.KafkaMessage {
```

Comment:
we can move this function to kafka.Message I guess and rename toKafkaMessage

Reply:
we cannot move kafka.Message there, it is a type belonging to the segmentio library.